File: Eniac.Reglas/FTP/BytesUploadedEventArgs.vb
Public Class BytesUploadedEventArgs
   Public Property FileName As String
   Public Property LoadedBytes As Long
   Public Property BufferSize As Long
   Public Property TotalBytes As Long
   Public Property Uri As Uri

   Public Sub New(fileName As String, uri As Uri, loadedBytes As Long, bufferSize As Long, totalBytes As Long)
      Me.FileName = fileName
      Me.Uri = uri
      Me.LoadedBytes = loadedBytes
      Me.BufferSize = bufferSize
      Me.TotalBytes = totalBytes
   End Sub

   Public Function Percentage() As Decimal
      Return (Convert.ToDecimal(LoadedBytes) / Convert.ToDecimal(TotalBytes)) * 100
   End Function
   Public Function PercentageAsInteger() As Integer
      Return Convert.ToInt32(Math.Truncate(Percentage))
   End Function
End Class

File: Eniac.Reglas/FTP/BytesUploadedEventArgs.cs
using System;

namespace AppFwk.Generic.Ftp
{
    public class BytesUploadedEventArgs : EventArgs
    {
        public BytesUploadedEventArgs()
        {

        }
        public BytesUploadedEventArgs(Int64 loadedBytes, Int64 bufferSize, Int64 totalBytes)
        {
            LoadedBytes = loadedBytes;
            BufferSize = bufferSize;
            TotalBytes = totalBytes;
        }
        public Int64 LoadedBytes { get; set; }
        public Int64 BufferSize { get; set; }
        public Int64 TotalBytes { get; set; }
        public Decimal Percentage
        {
            get
            {
                return (Convert.ToDecimal(LoadedBytes) / Convert.ToDecimal(TotalBytes)) * 100;
            }
        }
        public Int32 PercentageAsInteger
        {
            get
            {
                return Decimal.ToInt32(Math.Round(Percentage, 0));
            }
        }
    }
}